pkg/controller/jobs/leaderworkerset/leaderworkerset_controller.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leaderworkerset
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	leaderworkersetv1 "sigs.k8s.io/lws/api/leaderworkerset/v1"
+
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+)
+
+var (
+	gvk = leaderworkersetv1.SchemeGroupVersion.WithKind("LeaderWorkerSet")
+)
+
+const (
+	FrameworkName = "leaderworkerset"
+)
+
+func init() {
+	utilruntime.Must(jobframework.RegisterIntegration(FrameworkName, jobframework.IntegrationCallbacks{
+		SetupIndexes:   SetupIndexes,
+		NewReconciler:  NewPodReconciler,
```

Comment:
Any particular reason to introduce the Pod reconciler rather than LWS reconciler? I made a similar comment for StatefulSet, and I would prefer to have a LWS reconciler. 

One thing is performance - to avoid reconciling per pod, another reason is that with the LWS reconciler it might be simpler to create the prebuilt-workload by analyzing the LWS pod templates.

Reply:
> Any particular reason to introduce the Pod reconciler rather than LWS reconciler? 

This approach will not work well because the StatefulSet has an OwnerReference to the LeaderWorkerSet. When we remove the LeaderWorkerSet, it will be deleted before the StatefulSet and Pods, triggering the deletion of other objects.

Reply:
If we delete the LWS before the STS and Pods, we will lose context, and the Pods will remain stuck with a finalizer.